src/aws/flb_aws_util.c

```diff
@@ -81,6 +81,14 @@ char *flb_aws_endpoint(char* service, char* region)
 
 }
 
+char *removeProtocol (char *endpoint, char *protocol) {
+    if (strstr(endpoint, protocol)){
+        endpoint = endpoint + strlen(protocol);
+    }
+    return endpoint;
+}
+
```

Comment:
strstr checks substring... we want to check prefix.

See here: https://stackoverflow.com/questions/4770985/how-to-check-if-a-string-starts-with-another-string-in-c

I like the second answer using strncmp

Reply:
Oh yes! I see how using strstr is more error prone here. I will switch to strncmp.